am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java

```diff
@@ -13,127 +13,171 @@
+            .roleName(roleName)
             .resourceDefinition(ResourceDefinition.builder()
                 .serviceName(SERVICE_NAME)
                 .resourceType(RESOURCE_TYPE)
                 .resourceName(RESOURCE_NAME)
                 .build())
-            .attributePermissions(attributePermissionsForOtherRole)
+            .attributePermissions(attributePermissionsForRole)
             .build());
+
+        RolePermissions rolePermissions =
+            service.getRolePermissions(buildResource(SERVICE_NAME, RESOURCE_TYPE, RESOURCE_NAME), roleName);
+
+        assertThat(rolePermissions.getPermissions())
+            .hasSize(2)
+            .containsEntry(JsonPointer.valueOf("/child"), ImmutableSet.of(READ))
+            .containsEntry(JsonPointer.valueOf("/parent/age"), ImmutableSet.of(READ));
+
+        assertThat(rolePermissions.getSecurityClassifications())
+            .hasSize(2)
+            .containsEntry(JsonPointer.valueOf("/child"), PUBLIC)
+            .containsEntry(JsonPointer.valueOf("/parent/age"), PUBLIC);
+
+        assertThat(rolePermissions.getRoleAccessType()).isEqualTo(ROLE_BASED);
+
+        assertThat(rolePermissions.getRoleSecurityClassification()).isEqualTo(PUBLIC);
     }
 
     @Test
-    void returnListOfPermissionsForRoleName() {
-        Map<JsonPointer, Set<Permission>> accessRecord =
-            service.getRolePermissions(buildResource(SERVICE_NAME, RESOURCE_TYPE, RESOURCE_NAME), ROLE_NAMES);
+    void whenInsufficientRolePermissionsShouldReturnNull() {
+        Map.Entry<Set<Permission>, SecurityClassification> readPermission =
+            new Pair<>(ImmutableSet.of(READ), RESTRICTED);
 
-        assertThat(accessRecord)
-            .hasSize(3)
-            .containsEntry(JsonPointer.valueOf("/child"), ImmutableSet.of(READ))
-            .containsEntry(JsonPointer.valueOf("/parent/age"), ImmutableSet.of(CREATE))
-            .containsEntry(JsonPointer.valueOf("/address/street/line1"), ImmutableSet.of(CREATE));
+        attributePermissionsForRole = ImmutableMap.of(
+            JsonPointer.valueOf("/payment"), readPermission
+        );
+
+        importerService.addRole(roleName, IDAM, PUBLIC, ROLE_BASED);
+
+        importerService.grantDefaultPermission(DefaultPermissionGrant.builder()
+            .roleName(roleName)
+            .resourceDefinition(ResourceDefinition.builder()
+                .serviceName(SERVICE_NAME)
+                .resourceType(RESOURCE_TYPE)
+                .resourceName(RESOURCE_NAME)
+                .build())
+            .attributePermissions(attributePermissionsForRole)
+            .build());
+
+        RolePermissions rolePermissions =
+            service.getRolePermissions(buildResource(SERVICE_NAME, RESOURCE_TYPE, RESOURCE_NAME), roleName);
+
+        assertThat(rolePermissions).isNull();
+    }
+
+    @Test
+    void whenRolePermissionsHasSecurityClassificationShouldRemoveAttributesWithHigherSecurityClassification() {
```

Comment:
A `whenRolePermissionsHasSecurityClassification` part of method name is bit ambiguous. Maybe `whenGettingRolePermissionsShouldRemoveAttributesWithHigherSecurityClassification` be enough.